docs/user-guide/CAI.rst

```diff
@@ -0,0 +1,308 @@
+.. include:: /content/nemo.rsts
+
+.. _model-aligner-cai:
+
+Constitutional AI: Harmlessness from AI Feedback
+@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+
+`Constitutional AI (CAI)`_ is an approach by Anthropic to train AI systems that are helpful, honest, and harmless, without relying on human feedback labels for harmlessness.
+CAI uses a set of natural language principles to guide AI behavior and self-improvement.
+The method consists of two stages: a supervised learning stage, where the AI critiques and revises its own responses according to the principles, and a reinforcement learning stage, where the AI learns from its own feedback based on the principles.
+CAI allows training a harmless but non-evasive AI assistant that engages with harmful queries by explaining its objections to them.
+
+.. _Constitutional AI (CAI): https://arxiv.org/abs/2212.08073
+
+CAI
+###############
+The basic steps of CAI, as illustrated in Figure 1 of the paper (below):
+
+(Supervised Stage) Critique → Revision → Supervised Learning: The AI generates responses to harmfulness prompts using a helpful-only AI assistant, then critiques and revises its own responses according to a principle in the constitution, and then finetunes the original model on the revised responses.
+
+(RL Stage) AI Comparison Evaluations → Reward Model → Reinforcement Learning: The AI generates pairs of responses to harmfulness prompts using the finetuned model, then evaluates which response is better according to a principle in the constitution, and then trains a reward model based on this dataset of AI preferences and a human helpfulness preferences. The AI then trains with RL using the learned reward model.
+
```

Comment:
Add a figure title: Constitutional AI Basic Steps

Reply:
rst doesn't support figure titles. I have tried several workaround but failed getting it to render properly. I think that the first sentence in this paragraph is sufficient. thanks.

Reply:
OK, I understand. If we've run into the limitations of rst, then I would suggest removing the figure number altogether and just state "the figure below" or "the following figure." The same suggestion for the other figure in this file. That should work.